xtable-core/src/main/java/org/apache/xtable/parquet/ParquetSchemaExtractror.java

```diff
@@ -0,0 +1,486 @@
+                  .dataType(InternalType.STRING)
+                  .isNullable(false)
+                  .build())
+          .defaultValue("")
+          .build();
+  private static final ParquetSchemaConverter INSTANCE = new ParquetSchemaConverter();
+  private static final String ELEMENT = "element";
+  private static final String KEY = "key";
+  private static final String VALUE = "value";
+
+  public static parquetSchemaConverter getInstance() {
+    return INSTANCE;
+  }
+
+  public InternalSchema toInternalSchema(Schema schema) {
+    Map<String, IdMapping> fieldNameToIdMapping =
+        IdTracker.getInstance()
+            .getIdTracking(schema)
+            .map(
+                idTracking ->
+                    idTracking.getIdMappings().stream()
+                        .collect(Collectors.toMap(IdMapping::getName, Function.identity())))
+            .orElse(Collections.emptyMap());
+    return toInternalSchema(schema, null, fieldNameToIdMapping);
+  }
+
+  /**
+   * Converts the parquet {@link Schema} to {@link InternalSchema}.
+   *
+   * @param schema The schema being converted
+   * @param parentPath If this schema is nested within another, this will be a dot separated string
+   *     representing the path from the top most field to the current schema.
+   * @param fieldNameToIdMapping map of fieldName to IdMapping to track field IDs provided by the
+   *     source schema. If source schema does not contain IdMappings, map will be empty.
+   * @return a converted schema
+   */
+  private InternalSchema toInternalSchema(
+      Schema schema, String parentPath, Map<String, IdMapping> fieldNameToIdMapping) {
+    // TODO - Does not handle recursion in parquet schema
+    InternalType newDataType;
+    Map<InternalSchema.MetadataKey, Object> metadata = new HashMap<>();
+    switch (schema.getType()) {
+      case INT:
+        LogicalType logicalType = schema.getLogicalType();
+        if (logicalType instanceof LogicalTypes.Date) {
+          newDataType = InternalType.DATE;
+        } else {
+          newDataType = InternalType.INT;
+        }
+        break;
+      case STRING:
+        newDataType = InternalType.STRING;
+        break;
+      case BOOLEAN:
+        newDataType = InternalType.BOOLEAN;
+        break;
+      case BYTE_ARRAY:
+        logicalType = schema.getLogicalType();
+        // TODO: any metadata to add ?
+        if (logicalType== LogicalTypes.JSON) {
+          newDataType = InternalType.JSON;
+        }
+        else if (logicalType instanceof LogicalTypes.BSON) {
+          newDataType = InternalType.BSON;
+        }
+        else if (logicalType instanceof LogicalTypes.VARIANT) {
+          newDataType = InternalType.VARIANT;
+        }
+        else if (logicalType instanceof LogicalTypes.GEOMETRY) {
+          newDataType = InternalType.GEOMETRY;
+        }
+        else if (logicalType instanceof LogicalTypes.GEOGRAPHY) {
+          newDataType = InternalType.GEOGRAPHY;
```

Comment:
Would it to be simple to map it to `BYTES` ?

Reply:
Let me know your thoughts as well, I was asking because the targets (iceberg, delta and hudi) don't seem to support these types and just map it to byte array or binary.